packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -394,8 +387,8 @@ void _upgradeMap(Map<String, dynamic> map, ServiceObjectOwner owner) {
 }
 
 void _upgradeList(List<dynamic> list, ServiceObjectOwner owner) {
```

Comment:
Should/can this be `List<Object>` too, since you change `dynamic` to `Object` below?

Reply:
I think the JSON logic creates `List<dynamic>`s, not `List<Object>`s.